persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/ClinicalDataMyBatisRepository.java

```diff
@@ -21,32 +20,94 @@
+    public List<ClinicalData> getAllClinicalDataOfSampleInStudy(String studyId, String sampleId,
                                                                       String attributeId, String projection,
                                                                       Integer pageSize, Integer pageNumber,
                                                                       String sortBy, String direction) {
 
-        return clinicalDataMapper.getSampleClinicalData(studyIds, sampleIds, attributeId, projection, pageSize,
-                offsetCalculator.calculate(pageSize, pageNumber), sortBy, direction);
+        return clinicalDataMapper.getSampleClinicalData(Arrays.asList(studyId), Arrays.asList(sampleId), attributeId,
+                projection, pageSize, offsetCalculator.calculate(pageSize, pageNumber), sortBy, direction);
     }
 
     @Override
-    public BaseMeta getMetaSampleClinicalData(List<String> studyIds, List<String> sampleIds, String attributeId) {
-        return clinicalDataMapper.getMetaSampleClinicalData(studyIds, sampleIds, attributeId);
+    public BaseMeta getMetaSampleClinicalData(String studyId, String sampleId, String attributeId) {
+        return clinicalDataMapper.getMetaSampleClinicalData(Arrays.asList(studyId), Arrays.asList(sampleId),
+                attributeId);
     }
 
     @Override
-    public List<PatientClinicalData> getAllClinicalDataOfPatientInStudy(List<String> studyIds, List<String> patientIds,
+    public List<ClinicalData> getAllClinicalDataOfPatientInStudy(String studyId, String patientId,
                                                                         String attributeId, String projection,
                                                                         Integer pageSize, Integer pageNumber,
                                                                         String sortBy, String direction) {
 
-        return clinicalDataMapper.getPatientClinicalData(studyIds, patientIds, attributeId, projection, pageSize,
-                offsetCalculator.calculate(pageSize, pageNumber), sortBy, direction);
+        return clinicalDataMapper.getPatientClinicalData(Arrays.asList(studyId), Arrays.asList(patientId), attributeId,
+                projection, pageSize, offsetCalculator.calculate(pageSize, pageNumber), sortBy, direction);
     }
 
     @Override
-    public BaseMeta getMetaPatientClinicalData(List<String> studyIds, List<String> patientIds, String attributeId) {
-        return clinicalDataMapper.getMetaPatientClinicalData(studyIds, patientIds, attributeId);
+    public BaseMeta getMetaPatientClinicalData(String studyId, String patientId, String attributeId) {
+        return clinicalDataMapper.getMetaPatientClinicalData(Arrays.asList(studyId), Arrays.asList(patientId),
+                attributeId);
+    }
+
+    @Override
+    public List<ClinicalData> getAllClinicalDataInStudy(String studyId, String attributeId,
```

Comment:
passing in a clinicalDataType argument as a way to select between Sample and Patient makes sense if the code differences between the two modes of operation are minor, but below it seems like there is not much sharing of code between modes in the called methods --- the argument is tested and a different subroutine call is selected. If this is the case, maybe we should consider having two different methods here and dropping the clinicalDataType argument.

I guess it also depends on the callers of this function -- if there are very many calls, then making this change would push the "if" statement up into the calling code, which could clutter up the calling code in many places.